go/vt/binlog/keyrange_filter.go

```diff
@@ -39,12 +43,32 @@ func KeyRangeFilterFunc(keyrange *topodatapb.KeyRange, sendReply sendTransaction
 						continue
 					}
 				}
-				if !key.KeyRangeContains(keyrange, keyspaceID) {
-					// Skip keyspace ids that don't belong to the destination shard.
-					continue
+				if len(keyspaceIDS) == 1 {
+					if !key.KeyRangeContains(keyrange, keyspaceIDS[0]) {
+						// Skip keyspace ids that don't belong to the destination shard.
+						continue
+					}
+					filtered = append(filtered, statement)
+					matched = true
+				} else {
+					multipleQueries, err := splitMultiValueInsertQuery(string(statement.Sql), keyspaceIDS)
+					if err != nil {
+						continue
+					}
+					for rowNum, query := range multipleQueries {
+						if !key.KeyRangeContains(keyrange, keyspaceIDS[rowNum]) {
```

Comment:
We may need to bulk up these queries for efficiency. Basically, make splitMultiValueInsertQuery return a single sql that's applicable to the current keyrange.